src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1927,17 +1928,26 @@ internal ImmutableArray<Diagnostic> GetDiagnostics(CompilationStage stage, bool
                             var syntaxTree = syntaxTrees[i];
                             AppendLoadDirectiveDiagnostics(builder, _syntaxAndDeclarations, syntaxTree);
                             builder.AddRange(syntaxTree.GetDiagnostics(cancellationToken));
+                            if (parseOptionsReported.Add(syntaxTree.Options))
+                            {
+                                builder.AddRange(syntaxTree.Options.Errors);
```

Comment:
a couple of questions:
What if the same ParseOptions object was associated to multiple trees? use additional locations? which one takes precedence?
In VB, what if the same object is used in the CompilationOptions.ParseOptions as well?

---
In reply to: [105718150](https://github.com/dotnet/roslyn/pull/17113#discussion_r105718150) [](ancestors = 105718150)